src/UI/Components/TreeTable/TreeTable.test.tsx

```diff
@@ -73,6 +79,65 @@ test("TreeTable 1st level of nested property can be expanded", async () => {
   expect(screen.getByRole("row", { name: "Row-a$b" })).toBeVisible();
 });
 
+test("TreeTable with 1st level of attributes containing annotations should not render a value but be a link to the right tab.", async () => {
+  const serviceModel: ServiceModel = {
+    name: "service",
+    environment: "env",
+    attributes: [
+      {
+        name: "documentation",
+        type: "string",
+        description: "description",
+        modifier: "rw",
+        default_value: "",
+        default_value_set: false,
+        attribute_annotations: {
+          web_presentation: "documentation",
+          web_content_type: "text/markdown",
+          web_icon: "FaBook",
+          web_title: "Documentation",
+        },
+      },
+    ],
+    embedded_entities: [],
+    config: {},
+    lifecycle: {
+      initial_state: "initial",
+      states: [],
+      transfers: [],
+    },
+  };
+
+  // mock the setTab function
+  const setTab = jest.fn();
+
+  render(
+    inventorySetup(
+      {
+        candidate: null,
+        active: {
+          documentation: "```mermaid\ngraph LR\n    A --> B\n    B --> C\n```",
+        },
+        rollback: null,
+      },
+      serviceModel,
+    ),
+    setTab(),
+  );
+
+  // expect to find a row with a link to the documentation tab
+  const documentationButton = screen.getByRole("button", {
+    name: "documentation",
+  });
+  expect(documentationButton).toBeInTheDocument();
+
+  act(() => {
+    fireEvent.click(documentationButton);
+  });
+  // expect the setTab from the TreeTable component method to have been called
+  expect(setTab).toHaveBeenCalled();
```

Comment:
ui for the tabs is there, so assertion that the tab content will be visible should also work, right? If that's the case then it would be better to assert that instead just the call, but I can be mistaken as there are plenty of components, helpers, creators for instance services view

Reply:
Actually, the UI for the tabs is in a higher level. This TreeTable is inside a tab and only shows the attribute table with the collapsible/expandable attributes. Hence why the check if the setTab method is being called instead :)